// Copyright (c) 2024 Advanced Micro Devices, Inc. All Rights Reserved.
syntax = "proto3";

package Schola;

import "DType.proto";

message BoxPoint {
    repeated float values = 1;
    DType dtype = 2;
    repeated int32 shape = 3;
}

message DiscretePoint {
    int32 value = 1;
}

message MultiDiscretePoint {
    repeated int32 values = 1;
}

message MultiBinaryPoint {
    repeated bool values = 1;
}

message DictPoint {
    map<string, Point> values = 1;
}

message Point {
    oneof point {
        BoxPoint box_point = 1;
        DiscretePoint discrete_point = 2;
        MultiDiscretePoint multi_discrete_point = 3;
        MultiBinaryPoint multi_binary_point = 4;
        DictPoint dict_point = 5;
    }
}

